File: Перший курс/ІПЗ/Основи програмування (1 частина)/Код до звітів Основ програмування/лаб_6/ConsoleApplication1/ConsoleApplication1/func.cpp
#include "func.h"

double y_1(double x_1, double x_2, int n, double step) {
	while(x_1 <= x_2){
	double y = 0;
	if (x_1 < 0) {
	for (int i = 0; i <= (n - 1); i++) {
			int j = 0;
			for (j; j <= (n - 1); j++) {
					y += 1 / (x_1 - i + x_1 * j);
			}
		}
		return y;
	}
	else {
		double y = 0;
		cout << "division by zero (x) \t" << x_1 << endl;
		return y;
	}

	x_1 += step;
}  
}


double y_2(double x_1, double x_2, int n, double step) {
	while (x_1 <= x_2) {
		if (x_1 != 0) {
			double y = 1;
			for (int i = 1; i <= n; i++) {
				y *= (1 / x_1 - 1 / i);
			}
			return y;
		}
		else {
			double y = 0;
			cout << "division by zero (x)\t " << x_1 << endl;
			return y;
		}
		x_1 += step;
	}
}


bool checkValidInput(double x_1, double x_2, int n, double step)
{
	if ((n < 1) || (step <= 0) || (x_1 > x_2)) {
		cout << "data must be ((n >= 1) || (step > 0) || (x_1 <= x_2))" << endl;
		return true;
	}
	else {
		return false;
	}
}



	

File: Перший курс/ІПЗ/Основи програмування (2 частина)/Код до звітів Основ програмування(2 частина)/лаб_4/Task1/src/task/Func.java
package task;

import java.util.Scanner;

public class Func {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        System.out.print("Input x and n: \n");

        double x = scan.nextDouble();
        int n = scan.nextInt();
        if (n <= 1) {
            System.err.println("Incorrect data");
        } else {
            boolean a = true;
            double sum = 0;
            double mult = 1;
            double y = 0;
            label1:
            for (int i = 1; i <= (n - 1); i++) {
                    label:
                    for (int j = 0; j <= n; j++) {
                        if (i == j + x) {
                            System.out.println("y = " + y);
                            a = false;
                            break label1;
                        }
                        if (j + x == 0) {
                            System.err.println("Error. Division by zero");
                            a = false;
                            break label1;
                        }
                        sum += i / (j + x);
                    }
                    mult *= sum;


                System.out.println("y = " + mult);

            }
        }
    }
}